src/py4vasp/calculation/_partial_charge.py

```diff
@@ -0,0 +1,434 @@
+    truncate : float
+        The truncation of the Gaussian filter. The default is 3.0.
+    enhancement_factor : float
+        The enhancement factor for the output of the constant heigth
+        STM image. The default is 1000.
+    interpolation_factor : int
+        The interpolation factor for the z-direction in case of
+        constant current mode. The default is 10.
+    """
+
+    sigma_z: float = 4.0
+    sigma_xy: float = 4.0
+    truncate: float = 3.0
+    enhancement_factor: float = 1000
+    interpolation_factor: int = 10
+
+
+class PartialCharge(_base.Refinery, _structure.Mixin):
+    """Partial charges describe the fraction of the charge density in a certain energy,
+    band, or k-point range.
+
+    Partial charges are produced by a post-processing VASP run after self-consistent
+    convergence is achieved. They are stored in an array of shape
+    (ngxf, ngyf, ngzf, ispin, nbands, nkpts). The first three dimensions are the
+    FFT grid dimensions, the fourth dimension is the spin index, the fifth dimension
+    is the band index, and the sixth dimension is the k-point index. Both band and
+    k-point arrays are also saved and accessible in the .bands() and kpoints() methods.
+    If ispin=2, the second spin index is the magnetization density (up-down),
+    not the down-spin density.
+    Since this is postprocessing data for a fixed density, there are no ionic steps
+    to separate the data.
+    """
+
+    @property
+    def stm_settings(self):
+        return STM_settings()
+
+    @_base.data_access
+    def __str__(self):
+        """Return a string representation of the partial charge density."""
+        return f"""
+        {"spin polarized" if self._spin_polarized() else ""} partial charge density of {self._topology()}:
+        on fine FFT grid: {self.grid()}
+        {"summed over all contributing bands" if 0 in self.bands() else f" separated for bands: {self.bands()}"}
+        {"summed over all contributing k-points" if 0 in self.kpoints() else f" separated for k-points: {self.kpoints()}"}
+        """.strip()
+
+    @_base.data_access
+    def grid(self):
+        return self._raw_data.grid[:]
+
+    @_base.data_access
+    def to_dict(self):
+        """Store the partial charges in a dictionary.
+
+        Returns
+        -------
+        dict
+            The dictionary contains the partial charges as well as the structural
+            information for reference.
+        """
+
+        parchg = np.squeeze(self._raw_data.partial_charge[:].T)
+        return {
+            "structure": self._structure.read(),
+            "grid": self.grid(),
+            "bands": self.bands(),
+            "kpoints": self.kpoints(),
+            "partial_charge": parchg,
+        }
+
+    @_base.data_access
+    def to_stm(
+        self,
+        selection: str = "constant_height",
+        tip_height: float = 2.0,
+        current: float = 1.0,
+        supercell: Union[int, np.array] = 2,
+    ) -> Graph:
```

Comment:
Do you want users to be able to change the STM settings? If yes there are a couple of options to do that, you could either allow passing a settings object or take kwargs and create the settings object with it.

I would make tip_height and currect keyword only arguments. This avoids a call like `to_stm("total", 1.0)` where it is then hard to understand whether you set the height or the current. Another alternative would be to put these into the STM_settings but I assume you left them out because these are the most common parameters.
```suggestion
    @_base.data_access
    def to_stm(
        self,
        selection: str = "constant_height",
        *,
        tip_height: float = 2.0,
        current: float = 1.0,
        supercell: Union[int, np.array] = 2,
    ) -> Graph:
```

Reply:
Regarding the STM settings to be changed:

Yes, I think that is a good idea. At an earlier stage, I put the dataclass within the main class, allowing them to be set by the user. But this was changed to make the settings immutable in b0eb93c0c1c2f11e62f9ad58287cead473bfe9e7. Maybe we can put the contents of the dataclass in non-public attributes and provide setter and getter methods?

Regarding tip_height and current arguments:

This is a very good point. I did not know about keyword-only arguments. I think it's a good idea to use those. Indeed, putting them into the settings would be a worse user experience in my opinion.